services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/gdpr/GdprController.kt

```diff
@@ -157,9 +165,33 @@ class GdprController(
         }
     }
 
+    private fun authenticate(userId: String, token: Jwt, requiredScope: String) {
+        val sub = token.subject
+        val scopes: List<String>? = token.getClaimAsStringList(gdprProperties.authorizationField)
+
+        if (sub != userId) {
+            throw AuthenticationException("JWT sub was $sub, but user id in URL was $userId")
+        }
+        if (scopes == null || !scopes.contains(requiredScope)) {
+            throw AuthenticationException(
+                "JWT scopes were $scopes, which didn't include $requiredScope"
+            )
+        }
+    }
+
     class NotImplementedError(endpointName: String) :
```

Comment:
Could this be moved to for example ControllerExceptionHandler? This seems like a good fit for more general usage also.

Reply:
For general use, we probably want to return a HankeError code. Here we don't, since we're implementing an externally described API.

Reply:
Come to think of it, there's no real reason to check the disabled property in the controller anymore since SecurityFilter denies all access to the endpoints if disabled is true.

Reply:
Removed the disabled check and the NotImplemented stuff.